### ISSUE TITLE
Docs: Document code architecture and code reorg

### ISSUE THREAD

**Issue by adlrocha** [consensus-shipyard/ipc#445]
As the code base is growing in the number of different crates each with different roles and responsibilities, we should create a README describing the code architecture, the different crates that conform the agent, how they are used, and their interactions. This will help external contributors navigate the code, and find the right place the may want to introduce their change to. 

Additionally, as we do this exercise, we should consider a code reorganization to make the codebase cleaner and more accessible. E.g. we could divide the different handlers at https://github.com/consensus-shipyard/ipc-agent/tree/main/src/server/handlers by topic in different subcrates to prevent the number files to increase with the number of new commands in the agent. The commands [here](https://github.com/consensus-shipyard/ipc-agent/blob/5815f5955c1eaaf4bf07637e460cfbcee1f2bdf4/src/cli/commands/mod.rs#L76) could also be organized topically to make it more readable.

TL;DR, make our code base a charm to read and work with.
